pkg/resource/clickpipe.go

```diff
@@ -2047,27 +2146,32 @@ func (c *ClickPipeResource) syncClickPipeState(ctx context.Context, state *model
 			settingsModel.EnableFailoverSlots = types.BoolNull()
 		}
 
-		// Table mappings - preserve null values from state
-		var stateTableMappings []models.ClickPipePostgresTableMappingModel
+		// Table mappings - convert API response to Set (order doesn't matter)
+		// Get state mappings for preserving null values on optional fields
+		var stateTableMappingsMap map[string]models.ClickPipePostgresTableMappingModel
```

Comment:
The key here should be `struct{sourceSchemaName, sourceTable string}` (simple structs can be keys in Go) - each source table can only be present once, and I'm not sure if it's completely impossible to have dots in names